vm/parser.ts

```diff
@@ -711,6 +861,17 @@ module J2ME {
       }
     }
 
+    getClassNameSlow(): string {
+      return this._name || (this._name = ByteStream.readString(this.utf8Name));
+    }
```

Comment:
This (and all the other getters that are called often) should be memoizing, so they're only ever called once per instance:

```JS
    get className(): string {
      Object.defineProperty(this, "className", { value: ByteStream.readString(this.utf8Name) });
      return this.className;
    }
```